src/client/filter.rs

```diff
@@ -0,0 +1,292 @@
+            ObCompareOperator::GreaterOrEqualThan => ">=",
+            ObCompareOperator::NotEqual => "!=",
+            ObCompareOperator::Equal => "=",
+            ObCompareOperator::IsNull => "IS",
+            ObCompareOperator::IsNotNull => "IS_NOT",
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum FilterOp {
+    And = 0,
+    Or = 1,
+}
+
+pub struct ObTableFilterList {
+    pub op: FilterOp,
+    pub filters: Vec<Box<dyn Filter>>,
+}
+
+#[macro_export]
+macro_rules! filter_list {
+    ($op:expr, $($filter:expr),+ $(,)?) => {
+        ObTableFilterList {
+            op: $op,
+            filters: vec![$(Box::new($filter) as Box<dyn Filter>),+],
+        }
+    };
+}
+
+impl ObTableFilterList {
+    pub fn new<I>(op: FilterOp, filters: I) -> Self
+    where
+        I: IntoIterator<Item = Box<dyn Filter>>,
+    {
+        ObTableFilterList {
+            op,
+            filters: filters.into_iter().collect(),
+        }
+    }
+
+    pub fn add_filter(&mut self, filter: Box<dyn Filter>) {
+        self.filters.push(filter)
+    }
+
+    pub fn add_filters<I>(&mut self, filters: I)
+    where
+        I: IntoIterator<Item = Box<dyn Filter>>,
+    {
+        self.filters.extend(filters.into_iter())
+    }
+}
+
+impl Filter for ObTableFilterList {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn string(&self) -> String {
+        let string_op = match self.op {
+            FilterOp::And => " && ",
+            FilterOp::Or => " || ",
+        };
+
+        // Use an iterator with map and collect to efficiently concatenate strings
+        self.filters
+            .iter()
+            .map(|filter| {
+                let filter_string = filter.string();
+                if filter.as_any().is::<ObTableValueFilter>() {
+                    filter_string
+                } else {
+                    format!("({})", filter_string)
+                }
+            })
+            .collect::<Vec<_>>()
+            .join(string_op)
+    }
+
+    fn concat_string(&self, filter_string: &mut String) {
+        let string_op = match self.op {
+            FilterOp::And => " && ",
+            FilterOp::Or => " || ",
+        };
+
+        for (i, filter) in self.filters.iter().enumerate() {
+            if i != 0 {
+                filter_string.push_str(string_op);
+            }
+            if filter.as_any().is::<ObTableValueFilter>() {
+                filter.concat_string(filter_string);
+            } else {
+                filter_string.push('(');
+                filter.concat_string(filter_string);
+                filter_string.push(')');
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ObTableValueFilter {
+    pub op: ObCompareOperator,
+    pub column_name: String,
+    pub value: String,
+}
+
+impl ObTableValueFilter {
+    pub fn new<U: ToString, V: ToString>(op: ObCompareOperator, column_name: U, value: V) -> Self {
```

Comment:
`column_name` must be String type? And I guess `IntoString` may be a better type for value (no clone when the real type is string).

Reply:
Fixed